### DISCUSSION TITLE
Restoring the Win10 taskbar (the functioning tray area) on Win11 24H2 is very important

### DISCUSSION THREAD

**Discussion by Anixx** [valinet/ExplorerPatcher#3580] [General]
Restoring support for Win10 taskbar in 24H2 is very important, because it would affect also IoT Enterprise LTSC and Windows Server 2025. The IoT Enterprise LTSC has support term of 10 years.

I think, MS did not remove the Win10 taskbar from 24H2 intentionally, so, there should be a way to re-enable the system tray as well.

Also, the notes on Windows Server 2025 say that it boots into Win11 shell "by default", which suggests that maybe booting into Win10 shell is also possible and supported.

Win11 24H2 is the last version of Windows that has the Win10 taskbar, so it would be great if we could make it functional.

**Comment by milestprower92** [comment]
> Restoring support for Win10 taskbar in 24H2 is very important, because it would affect also IoT Enterprise LTSC and Windows Server 2025. The IoT Enterprise LTSC has support term of 10 years.
> 
> I think, MS did not remove the Win10 taskbar from 24H2 intentionally, so, there should be a way to re-enable the system tray as well.
> 
> Also, the notes on Windows Server 2025 say that it boots into Win11 shell "by default", which suggests that maybe booting into Win10 shell is also possible and supported.
> 
> Win11 24H2 is the last version of Windows that has the Win10 taskbar, so it would be great if we could make it functional.

Microsoft removed the WIndows 10 taskbar entirely on 24H2.
if you want to use the 10 taskbar on 24H2 install the latest prerelease (66.4) and use (Explorer Patcher's Implementation/recreation of the Windows 10 taskbar)

**Reply by Anixx** [reply]
I think, you are wrong, the taskbar is there on build 26100.

**Reply by milestprower92** [reply]
> I think, you are wrong, the taskbar is there on build 26100.

it was completely removed in a  Cumulative Update

**Reply by pyrates999** [reply]
Even if it still existed, MS added crash bombs to prevent the windows 10 taskbar from being used according to this: https://github.com/valinet/ExplorerPatcher/issues/3053#issuecomment-2077498668

**Reply by Anixx** [reply]
One can manipulate the features manually by using vivetool. There is no need to use EP to switch the features.

**Reply by pyrates999** [reply]
You can try to do this yourself if you wish.

**Reply by Anixx** [reply]
By the way, does the issue even exist at all? On 23H2 on my system the Win10 taskbar also has no tray area by default.

**Reply by pyrates999** [reply]
that's mentioned here: https://github.com/valinet/ExplorerPatcher/discussions/2363

**Reply by milestprower92** [reply]
> By the way, does the issue even exist at all? On 23H2 on my system the Win10 taskbar also has no tray area by default.

the 10 taskbar is fine on 22/23H2 (for now)

**Reply by milestprower92** [reply]
on Windows 11 24H2 insider builds microsoft started to remove the 10 taskbar

**Comment by Amrsatrio** [comment]
It is surprising that you do not believe the fact that the Windows 10 taskbar has been COMPLETELY removed in 26100.1297.  If you want the stock 10 taskbar back because Windows 10 (ExplorerPatcher) taskbar does not suit your needs, swapping explorer.exe with the version that still has the taskbar is the only way now. I don't know which one that has the system tray and works for 26100 release builds. There is no other way. Until the swapping method breaks due to API differences. Worth noting that I will NEVER make EP replace system files that are shipped with the system, that's yet another way to get 20-30 detections in VirusTotal, and also one of the ways to break Windows Update.

![image](https://github.com/user-attachments/assets/654a36ed-a338-48f2-9ece-b6b9c6423762)

If you do not believe me put that 2MB explorer.exe into IDA, download its symbols, and search for TrayUI:: in the functions view. It's gone. Gone.

**Comment by Amrsatrio** [comment]
@pyrates999 The crash bomb will always blow now since the old taskbar code has been removed. ViveTool is no longer an option for this without swapping files.

I'm locking this because you two seem to go nowhere in this discussion.
